laceworksdk/http_session.py

```diff
@@ -194,141 +199,191 @@ def _get_request_headers(self, org_access=False):
+        """
+        :param uri: uri to send the initial HTTP GET request to
+        :param params: dict of parameters for the HTTP request
+        :param kwargs: passed on to the requests package
 
-        logger.info(f"PATCH request to URI: {uri}")
-        logger.debug(f"PATCH request data:\n{data}")
+        :return: a generator that Gets and yields pages of data
 
-        # Perform a PATCH request
-        response = self._session.patch(uri, params=param, json=data, headers=self._get_request_headers(org_access=org))
+        :raises: ApiError if anything but expected response code is returned
+        """
 
-        # Validate the response
-        self._check_response_code(response, DEFAULT_SUCCESS_RESPONSE_CODES)
+        response = self.get(uri, params=params, **kwargs)
 
-        self._print_debug_response(response)
+        while True:
+            yield response
 
-        return response
+            try:
+                next_page = response.json()['paging']['urls']['nextPage']
+            except Exception:
+                next_page = None
 
-    def post(self, uri, org=False, data=None, param=None):
+            if next_page:
+                response = self.get(next_page, params=params, **kwargs)
+            else:
+                break
+
+    def get_items(self, uri, params=None, **kwargs):
         """
-        :param uri: uri to send the HTTP POST request to
-        :param org: boolean representing whether the request should be performed at the Organization level
-        :param data: json object containing the data
-        :param param: python object containing the parameters
+        :param uri: uri to send the initial HTTP GET request to
+        :param params: dict of parameters for the HTTP request
+        :param kwargs: passed on to the requests package
 
-        :return: response json
+        :return: a generator that GETs and yields items within pages of data
 
-        :raises: ApiError if unable to get a connection
+        :raises: ApiError if anything but expected response code is returned
+        :raises: MalformedResponse if the returned response does not contain a
+                top-level dictionary with an "data" key.
         """
 
-        self._check_access_token()
+        # Get generator for pages of JSON data
+        pages = self.get_pages(uri, params=params, **kwargs)
 
-        uri = f"{self._base_url}{uri}"
+        for page in pages:
+            page = page.json()
+            assert isinstance(page, dict)
 
-        logger.info(f"POST request to URI: {uri}")
-        logger.debug(f"POST request data:\n{data}")
+            items = page.get("data")
 
-        # Perform a POST request
-        response = self._session.post(uri, params=param, json=data, headers=self._get_request_headers(org_access=org))
+            if items is None:
+                error_message = f"'data' key not found in JSON data:\n{page}"
+                raise MalformedResponse(error_message)
+            else:
```

Comment:
no need for that else: statemement,... if items is none you raise an exception and therefore don't go further into the codebase